packages/lodestar/src/network/peers/utils.ts

```diff
@@ -65,24 +66,34 @@ export function findMissingSubnets(network: INetwork): number[] {
 }
 
 /**
- * If we're too close to maxPeer, we may want to disconnect some peers
+ * Synced: If we're too close to maxPeer, we may want to disconnect some peers
  * before connecting to new peers with missing subnets.
+ * Not synced: disconnect 50% worst/useless peers, keeps improving per epoch.
  * @param network
  * @param newPeer number of peers with new subnets to connect to
  * @param maxPeer max peer provided as option
+ * @param isSynced if the node is fully synced or not
  * @param reps our reputation store
  * @param retainRatio: 0 to 1, the ratio to consider disconnecting peers, 90% by default.
  */
 export function selectPeersToDisconnect(
   network: INetwork,
   newPeer: number,
   maxPeer: number,
+  isSynced: boolean,
   retainRatio = 0.9
 ): PeerId[] {
-  // TODO: prune useless peers
-  // peer that timeouts, that doesn't support required protocols (we still keep those as
-  // they might be useful to gossip) or that have bad gossip score later on
   const peers = network.getPeers({connected: true}).map((peer) => peer.id);
+  if (!isSynced) {
+    const syncPeers = getSyncPeers(network, undefined, network.getMaxPeer());
```

Comment:
is there a reason for not fetching peers directly so you can remove non sync peers as well??